### PR TITLE
[SSL for SaaS] add using dashboard for SSL for SaaS and change string processing in uploading_certificates.md

### PR DESCRIPTION
**Pull request by zechu-cf**
- Add docs for using the dashboard for SSL for SaaS.
- Shorten description of obtaining key and zone ID in ##Configure zone with proxy fallback origin since it's a repetition of ## API key and zone ID
- Fix certificate/private key string processing command in uploading-certificates.md . Previously it would cause the certificate to be unable to parse
- Other minor edits for clarity and consistency

### CODE REVIEW

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 1]
```diff
@@ -8,24 +8,26 @@ pcx-content-type: tutorial
 --------
 
 ## Configure zone with proxy fallback origin
-The fallback origin is used to route the traffic of your Custom Hostnames.  The fallback record is the value of the DNS record you set up via step 2 below.
+The fallback origin is where the traffic of your Custom Hostnames will be routed to.  The fallback record is the value of the DNS record you set up via step 2 below.
 
 1. Sign up your desired zone at [www.cloudflare.com](https://www.cloudflare.com) and select the Free plan.
-2. Add a DNS record pointing to the origin (fallback origin) for Cloudflare to send custom hostname traffic. This hostname is not provided to customers.  Name the hostname according to your naming convention.  For example: `proxy-fallback.saasprovider.com`.
+2. Under the **DNS** tab on the dashboard, add a DNS record pointing to the origin IP address (fallback origin) for Cloudflare to send custom hostname traffic. This hostname is not provided to customers.  Name the hostname according to your naming convention.  For example: `proxy-fallback.saasprovider.com`.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
1. Click **DNS**.
1. Add a DNS record pointing to the origin IP address (fallback origin) for Cloudflare to send custom hostname traffic. This hostname is not provided to customers.  Name the hostname according to your naming convention.  For example: `proxy-fallback.saasprovider.com`.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 2]
```diff
@@ -8,24 +8,26 @@ pcx-content-type: tutorial
...
 3. Set up an additional DNS CNAME record to serve as the CNAME target for your end customers.  CNAME this record to your fallback origin, and use a more user-friendly CNAME target for customers.  This record can optionally be a wildcard, e.g. `*.customers.saasprovider.com`.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
3. Set up an additional DNS CNAME record to serve as the CNAME target for your end customers. CNAME this record to your fallback origin and use a more user-friendly CNAME target for customers. This record can optionally be a wildcard, e.g. `*.customers.saasprovider.com`.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 3]
```diff
@@ -8,24 +8,26 @@ pcx-content-type: tutorial
...
-
-```bash
-$ curl -XPUT
-"https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/custom_hostnames/fallback_origin"\
--H "X-Auth-Email: {email}" -H "X-Auth-Key: {key}"\
--H "Content-Type: application/json"\
--d '{"origin":"proxy-fallback.saasprovider.com"}'
-```
+![Add a CNAME record](..//static/ssl-for-saas-dns.png)
+4. Upgrade your zone to an Enterprise plan, and contact your Customer Success Manager to enable SSL for SaaS Certificates.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
4. Upgrade your zone to an Enterprise plan and contact your Customer Success Manager to enable **SSL for SaaS Certificates**.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 4]
```diff
@@ -8,24 +8,26 @@ pcx-content-type: tutorial
...
+5. Set the **Fallback Origin** via either the dashboard or API.
+    * Via the dashboard: Go to **SSL/TLS** > **Custom Hostnames**, add your fallback origin defined in step 2 above and click **Add**.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
    * Via the dashboard: Go to **SSL/TLS** > **Custom Hostnames**, add your fallback origin defined in step 2 above, and click **Add**.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 5]
```diff
@@ -41,31 +43,46 @@ Each Custom Hostname requires successful Certificate Validation and Hostname Ver
...
 
 Once your account has been provisioned, you are ready to issue certificates. The call below will provision a request for certificates to be issued for `app.customer.com`, which represents your end customer.
 
-In this example, HTTP based validation is used ("method":"http") to issue this certificate. This requires HTTP traffic to be proxied through Cloudflare’s edge already, i.e., the CNAME from `app.customer.com` must be in place to your zone. If the CNAME is not yet in place, Cloudflare will ask its CA partner to retry until the request can be completed; see the [Validation Backoff Schedule](/ssl-for-saas/validation-backoff-schedule/) for specific timings.
+In this example, HTTP based validation is used ("method":"http") to issue this certificate. This requires HTTP traffic to be proxied through Cloudflare’s edge already, i.e., the CNAME from `app.customer.com` must be [in place to your zone](#setting-cname-at-customer-domain)). If the CNAME is not yet in place, Cloudflare will ask its CA partner to retry until the request can be completed; see the [Validation Backoff Schedule](/ssl-for-saas/validation-backoff-schedule/) for specific timings.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
In this example, HTTP based validation is used ("method":"http") to issue this certificate. This requires HTTP traffic to be proxied through Cloudflare’s edge already, i.e., the CNAME from `app.customer.com` must be [in place for your zone](#setting-cname-at-customer-domain)). If the CNAME is not yet in place, Cloudflare will ask its CA partner to retry until the request can be completed; see the [Validation Backoff Schedule](/ssl-for-saas/validation-backoff-schedule/) for specific timings.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 6]
```diff
@@ -41,31 +43,46 @@ Each Custom Hostname requires successful Certificate Validation and Hostname Ver
...
        -H "Content-Type: application/json"\
-       -d '{"hostname":"app.example.com", "ssl":{"method":"http","type":"dv"}}'
+       -d '{"hostname":"app.customer.com", "ssl":{"method":"http","type":"dv"}}'
 ```
 
 Note that it’s possible to serve these HTTP records from your own web servers, in advance of placing the CNAME. The payload returned includes the path where the CA will look for the challenge along with the body that should be returned.
 
+Alternatively, you may also issue certificates to custom hostnames via the dashboard. 
+1. Navigate to **SSL/TLS** > **Custom Hostnames** and click **Add Custom Hostname**.
+1. Add your customer's hostname `app.customer.com` and set the relevant options. Click **Add Custom Hostname**.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
1. Add your customer's hostname `app.customer.com` and set the relevant options. 
1. Click **Add Custom Hostname**.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 7]
```diff
@@ -41,31 +43,46 @@ Each Custom Hostname requires successful Certificate Validation and Hostname Ver
...
+![Add a custom hostname](..//static/ssl-for-saas-custom-hostname.png)
+1. You will be brought back to the previous screen, which will show “Pending” before it changes to “Active” after a while. If you see an error stating “custom hostname does not CNAME to this zone”, you need to [set the DNS record at the customer's domain](#setting-cname-at-customer-domain).
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]: Do we have anything more specific than `a while`? Like an upper bound on the time limit?

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 8]
```diff
@@ -41,31 +43,46 @@ Each Custom Hostname requires successful Certificate Validation and Hostname Ver
...
+
 Once domain validation has been completed, the certificates will be issued and distributed to Cloudflare’s edge. With a CNAME in place, the entire process—from validation to issuance to edge deployment—completes in approximately 90 seconds.
+
+--------
+
+## Setting CNAME at customer domain
+Your customer needs to set up a CNAME record at their DNS provider, pointing to your CNAME target configured in step 3 [above](#configure-zone-with-proxy-fallback-origin). For example:
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
Your customer needs to set up a CNAME record at their DNS provider, pointing to your CNAME target configured in [a previous step](#configure-zone-with-proxy-fallback-origin). For example:
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 9]
```diff
@@ -8,24 +8,26 @@ pcx-content-type: tutorial
 --------
 
 ## Configure zone with proxy fallback origin
-The fallback origin is used to route the traffic of your Custom Hostnames.  The fallback record is the value of the DNS record you set up via step 2 below.
+The fallback origin is where the traffic of your Custom Hostnames will be routed to.  The fallback record is the value of the DNS record you set up via step 2 below.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
The fallback origin is where the traffic of your Custom Hostnames will be routed to.  The fallback record is the value of the DNS record you set up via step 2.
```

File: products/ssl/src/content/ssl-for-saas/index.md [hunk 10]
```diff
@@ -5,8 +5,8 @@ pcx-content-type: landing-page
 
 # SSL for SaaS
 
-Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains.  Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
-Once the CNAME is added, a single API call to issue a SSL for SaaS certificate initiates domain validation with a Certificate Authority (CA) and issuance of two SSL certificates:
+Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains.  For example, a customer may want to use their “vanity” domain ``app.customer.com`` to point to an application hosted on your Cloudflare zone ``service.saas.com``. Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/index.md]:
```suggestion
Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains.  For example, a customer may want to use their “vanity” domain `app.customer.com` to point to an application hosted on your Cloudflare zone `service.saas.com`. Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
```

**Review by kodster28 — CHANGES_REQUESTED** [review]

File: products/ssl/src/content/ssl-for-saas/getting-started.md (hunk 7 above)
Comment (zechu-cf) [reply products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
1. You will be brought back to the previous screen, which will show “Pending” before it changes to “Active” within 5 minutes. If you see an error stating “custom hostname does not CNAME to this zone”, you need to [set the DNS record at the customer's domain](#setting-cname-at-customer-domain).
```
I've changed it to within 5 minutes.

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 11]
```diff
@@ -8,24 +8,27 @@ pcx-content-type: tutorial
 --------
 
 ## Configure zone with proxy fallback origin
-The fallback origin is used to route the traffic of your Custom Hostnames.  The fallback record is the value of the DNS record you set up via step 2 below.
+The fallback origin is where the traffic of your Custom Hostnames will be routed to.  The fallback record is the value of the DNS record you set up via step 2.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
The fallback origin is where the traffic of your Custom Hostnames will be routed to. The fallback record is the value of the DNS record you set up via step 2.
```

File: products/ssl/src/content/ssl-for-saas/getting-started.md [hunk 12]
```diff
@@ -8,24 +8,27 @@ pcx-content-type: tutorial
...
-
-```bash
-$ curl -XPUT
-"https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/custom_hostnames/fallback_origin"\
--H "X-Auth-Email: {email}" -H "X-Auth-Key: {key}"\
--H "Content-Type: application/json"\
--d '{"origin":"proxy-fallback.saasprovider.com"}'
-```
+1. Click **DNS**.
+1. Add a DNS record pointing to the origin IP address (fallback origin) for Cloudflare to send custom hostname traffic. This hostname is not provided to customers.  Name the hostname according to your naming convention.  For example: `proxy-fallback.saasprovider.com`.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/getting-started.md]:
```suggestion
1. Add a DNS record pointing to the origin IP address (fallback origin) for Cloudflare to send custom hostname traffic. This hostname is not provided to customers. Name the hostname according to your naming convention. For example: `proxy-fallback.saasprovider.com`.
```

File: products/ssl/src/content/ssl-for-saas/index.md [hunk 13]
```diff
@@ -5,8 +5,8 @@ pcx-content-type: landing-page
 
 # SSL for SaaS
 
-Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains.  Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
-Once the CNAME is added, a single API call to issue a SSL for SaaS certificate initiates domain validation with a Certificate Authority (CA) and issuance of two SSL certificates:
+Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains.  For example, a customer may want to use their “vanity” domain `app.customer.com` to point to an application hosted on your Cloudflare zone `service.saas.com`. Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
```
Comment (kodster28) [inline products/ssl/src/content/ssl-for-saas/index.md]:
```suggestion
Cloudflare [SSL for SaaS](https://www.cloudflare.com/saas/) extends the security and performance benefits of Cloudflare’s network to your customers via their own custom or “vanity” domains. For example, a customer may want to use their “vanity” domain `app.customer.com` to point to an application hosted on your Cloudflare zone `service.saas.com`. Issuing certificates requires minimal interaction from your customer to add a CNAME from their custom hostname to your domain.
```

**Review by kodster28 — APPROVED** [review]

**Pull request merged**